SDALib-CPP/SDAReplay/ReplayMain.cpp

```diff
@@ -1,11 +1,18 @@
 #include "ReplayDriver.h"
 
-int main()
+int main(int p_argc, char **p_path)
 {
-    std::string replayFileName("ReplayData/bbRecording20220330-123618.txt");
+    std::string replayFileName;  // initialize the string
+    for (int i = 1; i < p_argc; i++)
+    {
+        // picks the argument written after --bbfile
+        if (strcmp(p_path[i], "--bbfile") == 0) continue;
+
+        replayFileName = p_path[++i];
```

Comment:
Would be better to do something like 
```cpp
if (strcmp(p_path[i], "--bbfile") == 0)
{
    replayFileName = p_path[++i];
}
```
This would allow for more arguments to be added later. However this is not safe (just like your current way of doing it). In the final loop, you have i = p_argc - 1. Reading from this index is fine, but if the value is not "--bbfile", you will continue to line 11. Here you do ++i, meaning i now equals p_argc. That means you read outside of the array. Maybe do something like 
```cpp
    for (int i = 1; i < p_argc - 1; i++)
    {
        if (strcmp(p_path[i], "--bbfile") == 0)
        {
            replayFileName = p_path[++i];
        }
    }
```